api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/ActionPool.java

```diff
@@ -0,0 +1,99 @@
+package edu.cornell.mannlib.vitro.webapp.dynapi;
+
+import static edu.cornell.mannlib.vitro.webapp.modelaccess.ModelNames.FULL_UNION;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.ServletContext;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.jena.ontology.OntModel;
+
+import edu.cornell.mannlib.vitro.webapp.dynapi.components.Action;
+import edu.cornell.mannlib.vitro.webapp.dynapi.components.DefaultAction;
+import edu.cornell.mannlib.vitro.webapp.modelaccess.ContextModelAccess;
+import edu.cornell.mannlib.vitro.webapp.modelaccess.ModelAccess;
+import edu.cornell.mannlib.vitro.webapp.utils.configuration.ConfigurationBeanLoader;
+
+public class ActionPool {
+
+	private static ActionPool INSTANCE = null;
+ 	private static final Log log = LogFactory.getLog(ActionPool.class);
+ 	
+	private Map<String,Action> actions;
+	private ServletContext ctx;
+	private ConfigurationBeanLoader loader;
+	private ContextModelAccess modelAccess;
+	private OntModel dynamicAPIModel;
+
+	private ActionPool(){
+		this.actions = new HashMap<String,Action>();
+		INSTANCE = this;
+	}
+	
+	public static ActionPool getInstance() {
+		if (INSTANCE == null) {
+			INSTANCE = new ActionPool();
+		}
+		return INSTANCE;
+	}
+	
+	public Action getByName(String name) {
+		Action action = actions.get(name);
+		if (action == null) {
+			action = new DefaultAction();
```

Comment:
May want to discuss the value of a default action. This action as is will not have a `rpc` or `firstStep` assigned as it is not being loaded.

Reply:
DefaultAction is a null object to provide behavior if no action is found.

Reply:
Why return an action at all if there are none? What is the expected behavior of the DefaultAction?